airflow/executors/celery_executor.py

```diff
@@ -142,7 +146,11 @@ def __init__(self):
             self._sync_parallelism = max(1, cpu_count() - 1)
         self.bulk_state_fetcher = BulkStateFetcher(self._sync_parallelism)
         self.tasks = {}
-        self.last_state = {}
+        # Mapping of tasks we've adopted, ordered by the earliest date they timeout
+        self.adopted_task_timeouts: Dict[TaskInstanceKey, datetime.datetime] = OrderedDict()
+        self.adoption_timeout = datetime.timedelta(
+            seconds=conf.getint('celery', 'adopted_task_timeout', fallback=600)
```

Comment:
Only one more minor comment relating to naming: `self. adopted_task_timeouts` and `adopted_task_timeout` may make this less comprehensive.

I understand they are linked via `self.adopted_task_timeouts[ti.key] = ti.queued_dttm + self.adoption_timeout`, but the extremely similar names made me think they are "directly" linked to each other.

I would suggest to rename either one.

Otherwise all look good to me.

Reply:
Good shout. I'll rename `adopted_task_timeouts` to `adopted_task_by_timeout`? `abandon_adopted_tasks_after`?

Reply:
Or perhaps `adopted_task_timeout` to `task_adoption_timeout` (and leave `adopted_task_timeouts`)

Reply:
I leave this to an English speaker 😉  I will be happy as long as they distinguish well from each other. 

But my preference goes for `adopted_task_timeout`-> `task_adoption_timeout`